src/ucm/cuda/cudamem.c

```diff
@@ -116,7 +142,8 @@ static void ucm_cudafree_dispatch_events(void *dptr)
     }
     ucs_assert(dptr == (void *)pbase);
 
-    ucm_dispatch_mem_type_free((void *)dptr, psize, UCS_MEMORY_TYPE_CUDA);
+    ucm_dispatch_mem_type_free((void *)dptr, psize,
+                               ucm_cuda_get_mem_type(dptr));
```

Comment:
we were not caring about the type of cuda memory in the cache while freeing it?  just detecting the cache entry for that address. do we need this?

Reply:
@bureddy 

> we were not caring about the type of cuda memory in the cache while freeing it?

No, but PR that fixes MemType Cache has assert() to verify it. So, it fails w/o this fix